crates/blockifier/src/execution/native/syscall_handler.rs

```diff
@@ -83,14 +85,74 @@ impl<'state> NativeSyscallHandler<'state> {
             accessed_storage_keys: HashSet::new(),
         }
     }
+
+    pub fn execute_inner_call(
+        &mut self,
+        entry_point: CallEntryPoint,
+        remaining_gas: &mut u128,
+    ) -> SyscallResult<CallInfo> {
+        let call_info = entry_point
+            .execute(self.state, self.execution_resources, self.execution_context)
+            .map_err(|e| encode_str_as_felts(&e.to_string()))?;
+        let retdata = call_info.execution.retdata.0.clone();
+
+        if call_info.execution.failed {
+            // In VM it's wrapped into `SyscallExecutionError::SyscallError`
+            return Err(retdata);
+        }
+
+        self.update_remaining_gas(remaining_gas, &call_info);
+
+        self.inner_calls.push(call_info.clone());
+
+        Ok(call_info)
+    }
```

Comment:
Doesn't this function exists already?

Reply:
This function exists, but it was better to implement it one more time since: - it pushes to inner_calls
- it handles errors in the different way
- it works with `&mut u128`, but not with `&mut u64`